sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -392,13 +393,23 @@ export class MessageReceiver extends LinkEntity {
                 bMessage.messageId as string,
                 setTimeout(async () => {
                   try {
+                    let receiverName;
+                    if (this._receiver) {
+                      receiverName = this._receiver.name;
+                    } else {
+                      throw new Error(
```

Comment:
We would enter the `else` block here in one of the below conditions
- User closed the receiver/queueClient/subscriptionClient/serviceBusClient
- There was an error on the AMQP receiver link causing `_onAmqpError` followed by `_onAmqpClose` to be called
- There was an error on the AMQP session causing `_onSessionError`  followed by `_onSessionClose` to be called

In the 2nd and 3rd scenarios, we call `this._clearAllMessageLockRenewTimers();` to clear the renewlock loops
In the first case, we don't but we should.

In the first case, user has explicitly closed entities. Throwing an error back to the user at this point is not what the user expects. They are done with the receiver, and they don't care if there was an ongoing lock renewal going on or not.

The second and third cases are when receiver closed due to an error in which case we call the user's error handler appropriately. Therefore, raising another error that the lock renewal cannot happen is not helpful. There is a bigger issue with the receiver link and that has been reported using the user's error handler.

Therefore, instead of throwing an error here, we should simply exit the auto renewal loop.

Reply:
For second and third case, yes we do call user's onError handler eventually. But the right error message must surface up right? This throwing only appends onto the error stack trace in containing try catch, doesn't throw a separate error as such.

As for first case, is it an issue that's already being tracked? I suppose that involves invoking `._clearAllMessageLockRenewTimers()` on `close()` for both sessions and non-sessions and running all the lock renewal tests? Maybe even add a test that makes sure this error doesn't surface when receiver is explicitly closed.

Reply:
> For second and third case, yes we do call user's onError handler eventually. But the right error message must surface up right? 

The right error message is that the receiver got closed along with any accompanying error. This is done by the `_onAmqpError` or `_onSessionError` when they call the user's onError handler

The fact that we (the sdk) try to continue to attempt to renewLock is the sdks fault and not the user. There is nothing the user can do more with this error for renewLock than what they can do with the error from the onError handler.

> As for first case, is it an issue that's already being tracked?

No, we found this as part of the code review in this PR

> I suppose that involves invoking ._clearAllMessageLockRenewTimers() on close() for both sessions and non-sessions and running all the lock renewal tests?

Yes, we should call `_clearAllMessageLockRenewTimers()` on `close()` in `messageReceiver.ts` file
No, we don't need to do this for sessions, as it is already taken care of. See https://github.com/Azure/azure-sdk-for-js/blob/%40azure/service-bus_1.0.0-preview.2/sdk/servicebus/service-bus/src/session/messageSession.ts#L486

> Maybe even add a test that makes sure this error doesn't surface when receiver is explicitly closed

Since, I still believe that we don't need to throw an error when receiver cannot be found during auto-lock renewal, no such test is required

Reply:
Ok, removed.
Also, added fix to invoke `_clearAllMessageLockRenewTimers()` as discussed.